configs/projects/puppet-agent.rb

```diff
@@ -2,19 +2,46 @@
   platform = proj.get_platform
 
   # Project level settings our components will care about
-  # Windows has its own separate layout
   if platform.is_windows?
-    # Our install root can't have spaces in it, so we take advantage of short cuts.
-    # However, in order to take advantage of these shortcuts, we need to explicitly
-    # create the directories before using any shortcuts
-    proj.directory "C:/Program Files/Puppet Labs"
-    proj.setting(:install_root, "C:/Progra~1/Puppet~1")
-    proj.setting(:sysconfdir, "C:/ProgramData/PuppetLabs")
+    proj.setting(:company_name, "Puppet Labs")
+    proj.setting(:company_id, "PuppetLabs")
+    proj.setting(:common_product_id, "PuppetInstaller")
+    proj.setting(:product_id, "Puppet")
+    proj.setting(:upgrade_code, "2AD3D11C-61B3-4710-B106-B4FDEC5FA358")
+    if platform.architecture == "x64"
+      proj.setting(:product_name, "Puppet Agent (64-bit)")
+      proj.setting(:win64, "yes")
+      proj.setting(:base_dir, "ProgramFiles64Folder")
+      proj.setting(:RememberedInstallDirRegKey, "RememberedInstallDir64")
+    else
+      proj.setting(:product_name, "Puppet Agent")
+      proj.setting(:win64, "no")
+      proj.setting(:base_dir, "ProgramFilesFolder")
+      proj.setting(:RememberedInstallDirRegKey, "RememberedInstallDir")
+    end
+    proj.setting(:links, {
+        :HelpLink => "http://links.puppetlabs.com/customer-support-foss",
+        :CommunityLink => "http://links.puppetlabs.com/windows-installer-starting-out",
+        :ForgeLink => "http://links.puppetlabs.com/forge-windows",
+        :NextStepLink => "http://links.puppetlabs.com/windows-installer-next-steps-foss",
+        :ManualLink => "http://links.puppetlabs.com/windows-manual-foss",
+      })
+    proj.setting(:UI_exitdialogtext, "Manage your first resources on this node, explore the Puppet community and get support using the shortcuts in the Documentation folder of your Start Menu.")
+
+    # Directory IDs
+    proj.setting(:bindir_id, "bindir")
+
+    # We build for windows not in the final destination, but in the paths that correspond
+    # to the directory ids expected by WIX. This will allow for a portable installation (ideally).
+    proj.setting(:install_root, File.join("C:", proj.base_dir, proj.company_id))
```

Comment:
So we need a function to generate the short version of the name, and to create the directory as before? Or going a different route for MSI staging?